### PR TITLE
[now-cli] Add tests for project linking and framework detection

### PR DESCRIPTION
**Pull request by lucleray**
Adds two test cases:
- linking a project and overwriting build settings
- "overwrite settings" prompt should not appear when no framework is detected

### CODE REVIEW

File: packages/now-cli/test/integration.js [hunk 1]
```diff
@@ -2237,6 +2250,119 @@ test('ensure `github` and `scope` are not sent to the API', async t => {
...
+  now.stdin.write('no\n');
+
+  await waitForPrompt(now, chunk =>
+    chunk.includes('What’s your project’s name?')
+  );
+  now.stdin.write(`${projectName}\n`);
+
+  await waitForPrompt(now, chunk => {
+    if (chunk.includes('Want to override the settings?')) {
+      throw new Error();
```
Comment (styfle) [inline packages/now-cli/test/integration.js]: ~~throw new Error('Should not ask to override the settings');~~

Comment (styfle) [reply packages/now-cli/test/integration.js]: Actually could we use ava's assert to say "is not"?

File: packages/now-cli/test/helpers/prepare.js [hunk 2]
```diff
@@ -477,6 +477,15 @@ CMD ["node", "index.js"]`,
         },
       }),
     },
+    'project-link': {
+      'pages/index.js':
+        'export default () => <div><h1>Now CLI test</h1><p>Zero-config + Next.js</p></div>',
```
Comment (styfle) [inline packages/now-cli/test/helpers/prepare.js]: Why does this say Next.js but the dependency is gatsby?

Comment (lucleray) [reply packages/now-cli/test/helpers/prepare.js]:
```suggestion
        'export default () => <div><h1>Now CLI test</h1></div>',
```

File: packages/now-cli/test/integration.js (hunk 1 above)
Comment (lucleray) [reply packages/now-cli/test/integration.js]: @styfle I updated it to use `t.notThrowsAsync`

File: packages/now-cli/test/integration.js [hunk 3]
```diff
@@ -148,6 +148,28 @@ const apiFetch = (url, { headers, ...options } = {}) => {
...
+        }
+      } catch (error) {
+        reject(error);
+      }
+    };
+
+    cp.stdout.on('data', listener);
+    cp.stderr.on('data', listener);
+
+    cp.on('exit', () => resolve());
```
Comment (lucleray) [inline packages/now-cli/test/integration.js]:
```suggestion
    cp.on('exit', () => reject());
```

File: packages/now-cli/test/integration.js (hunk 1 above)
Comment (styfle) [reply packages/now-cli/test/integration.js]: @lucleray But why throw at all?

Instead you could do:

```js
t.false(chunk.includes('Want to override the settings?'), 'Should not ask to override');
```

https://github.com/avajs/ava/blob/master/docs/03-assertions.md

Comment (lucleray) [reply packages/now-cli/test/integration.js]: I like it! Updated 👍

**Review by styfle — APPROVED** [review]
🏅

**Pull request merged**